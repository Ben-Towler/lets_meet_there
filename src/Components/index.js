import SiteHeader from './SiteHeader/SiteHeader.component';
import SearchForm from './SearchForm/SearchForm.component';
import Trip from './Trip/Trip.component';
import Button from './Button/Button.component';
import Form from './Form/Form.component';
import Flight from './Flight/Flight.component';
import FlightSummary from './FlightSummary/FlightSummary.component';
import BookingLink from './BookingLink/BookingLink.component';
import Spinner from './Spinner/Spinner.component';
import LocationSearch from './LocationSearch/locationSearch.component.tsx';

export {
  SiteHeader,
  BookingLink,
  SearchForm,
  Trip,
  Flight,
  FlightSummary,
  Button,
  Form,
  Spinner,
  LocationSearch
}