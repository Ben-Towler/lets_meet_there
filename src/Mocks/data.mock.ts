export const placeMocks = {
    65698: {
      CityId: 'LOND',
      CityName: 'London',
      CountryName: 'United Kingdom',
      IataCode: 'LHR',
      Name: 'London Heathrow',
      PlaceId: 65698,
      SkyscannerCode: 'LHR',
      Type: 'Station'
    }
  };

export const quotes = [{
  QuoteId: 1,
  MinPrice: 349,
  Direct: false,
  OutboundLeg: {
    CarrierIds: [
      1090
    ],
    OriginId: 65698,
    DestinationId: 65698,
    DepartureDate: '2020-06-26T00:00:00'
  },
  InboundLeg: {
    CarrierIds: [
      1755
    ],
    OriginId: 65698,
    DestinationId: 65698,
    DepartureDate: '2020-06-29T00:00:00'
  },
  QuoteDateTime: '2020-06-17T21:12:00'
}];

export const carriers = [{

}];

export const flightOne = {
  Direct: false,
  InboundLeg: {
    CarrierIds: [1324],
    DepartureDate: '2020-06-27T00:00:00',
    DestinationId: 65698,
    OriginId: 65698,
  },
  MinPrice: 213,
  OutboundLeg: {
    CarrierIds: [1324],
    DepartureDate: '2020-06-25T00:00:00',
    DestinationId: 65698,
    OriginId: 65698,
  },
  QuoteDateTime: '2020-06-17T23:03:00',
  QuoteId: 19
}

export const flightTwo = {
  Direct: false,
  InboundLeg: {
    CarrierIds: [1324],
    DepartureDate: '2020-06-27T00:00:00',
    DestinationId: 65698,
    OriginId: 65698
  },
  MinPrice: 192,
  OutboundLeg: {
    CarrierIds: [1324],
    DepartureDate: '2020-06-25T00:00:00',
    DestinationId: 65698,
    OriginId: 65698
  },
  QuoteDateTime: '2020-06-15T19:17:00',
  QuoteId: 10
}

export const flightListMock = [[
  66328,
  [
    {
      QuoteId: 59,
      MinPrice: 177,
      Direct: false,
      OutboundLeg: {
        CarrierIds: [
          1090
        ],
        OriginId: 65698,
        DestinationId: 65698,
        DepartureDate: '2020-06-26T00:00:00'
      },
      InboundLeg: {
        CarrierIds: [
          1090
        ],
        OriginId: 65698,
        DestinationId: 65698,
        DepartureDate: '2020-06-29T00:00:00'
      },
      QuoteDateTime: '2020-06-18T10:10:00'
    }
  ], [
    {
      QuoteId: 9,
      MinPrice: 26,
      Direct: true,
      OutboundLeg: {
        CarrierIds: [
          1090
        ],
        OriginId: 65698,
        DestinationId: 65698,
        DepartureDate: '2020-06-26T00:00:00'
      },
      InboundLeg: {
        CarrierIds: [
          1090
        ],
        OriginId: 65698,
        DestinationId: 65698,
        DepartureDate: '2020-06-29T00:00:00'
      },
      QuoteDateTime: '2020-06-18T16:38:00'
    }
  ]
]];

export const favLocation = {
  city: 'Berlin',
  country: 'Germany'
}

export const userCity = 'Berlin';
export const friendCity = 'Berlin';

export const searchDetailsForRemoveHandler = {
  origin: '',
  destination: '',
  outboundDate: '',
  inboundDate: ''
}