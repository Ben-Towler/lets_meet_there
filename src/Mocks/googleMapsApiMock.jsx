export const setupGoogleMock = () => {
  // Mock Google Maps JavaScript API 
  const google = {
    maps: {
      places: {
        Autocomplete: class {},
        AutocompleteService:class{},
        PlacesServiceStatus: {
          INVALID_REQUEST: 'INVALID_REQUEST',
          NOT_FOUND: 'NOT_FOUND',
          OK: 'OK',
          OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
          REQUEST_DENIED: 'REQUEST_DENIED',
          UNKNOWN_ERROR: 'UNKNOWN_ERROR',
          ZERO_RESULTS: 'ZERO_RESULTS',
        },
      },
      Geocoder:class{},
      GeocoderStatus: {
        ERROR: 'ERROR',
        INVALID_REQUEST: 'INVALID_REQUEST',
        OK: 'OK',
        OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
        REQUEST_DENIED: 'REQUEST_DENIED',
        UNKNOWN_ERROR: 'UNKNOWN_ERROR',
        ZERO_RESULTS: 'ZERO_RESULTS',
      },
    },
  };
  global.window.google = google;
};

