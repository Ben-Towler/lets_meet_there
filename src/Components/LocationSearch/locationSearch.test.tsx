import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {LocationSearch} from 'Components';
import { setupGoogleMock } from '../../Mocks/googleMapsApiMock';

configure({adapter: new Adapter()});

beforeAll(() => {
  setupGoogleMock();
});

describe('Component: LocationSearch', () => {

  test.only('it should render correctly', () => {
    const locationSearchTree = renderer
    .create(<LocationSearch/>)
    .toJSON();
  expect(locationSearchTree).toMatchSnapshot();
  });

});