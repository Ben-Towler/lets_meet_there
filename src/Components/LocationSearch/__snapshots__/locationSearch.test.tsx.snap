// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LocationSearch it should render correctly 1`] = `
<div>
  <div
    className="autocomplete-root"
  >
    <input
      aria-autocomplete="list"
      aria-expanded={false}
      autoComplete="off"
      className="form-control"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      role="combobox"
      type="text"
      value=""
    />
    <div
      className="autocomplete-dropdown-container"
    />
  </div>
</div>
`;
